modulemd/include/private/glib-extensions.h

```diff
@@ -14,10 +14,46 @@
 #pragma once
 
 #include <glib.h>
+#include <glib/gtypes.h>
 
 #include "config.h"
 
 /* GDate autoptr cleanup was finally added in GLib 2.63.3. */
 #ifndef HAVE_GDATE_AUTOPTR
 G_DEFINE_AUTOPTR_CLEANUP_FUNC (GDate, g_date_free)
 #endif
+
+#ifndef HAVE_EXTEND_AND_STEAL
+
+#ifndef MIN_ARRAY_SIZE
+#define MIN_ARRAY_SIZE 16
+#endif
+
+
+void
+g_ptr_array_extend (GPtrArray *array_to_extend,
+                    GPtrArray *array,
+                    GCopyFunc func,
+                    gpointer user_data);
+
+
+/*
+ * g_ptr_array_extend_and_steal:
+ * @array_to_extend: (transfer none): a #GPtrArray.
+ * @array: (transfer container): a #GPtrArray to add to the end of
+ *     @array_to_extend.
+ *
+ * Adds all the pointers in @array to the end of @array_to_extend, transferring
+ * ownership of each element from @array to @array_to_extend and modifying
+ * @array_to_extend in-place. @array is then freed.
+ *
+ * As with g_ptr_array_free(), @array will be destroyed if its reference count
+ * is 1. If its reference count is higher, it will be decremented and the
+ * length of @array set to zero.
+ *
+ * Since: 2.62
```

Comment:
```suggestion
 * Since: GLib-2.62
```
I agree with making this a regular comment instead of GtkDoc text. But I still suggest tagging the "Since:" as belonging to GLib.